File: Python3/Repetitivas/ej3.py
# -*- coding: utf-8 -*-

#Algoritmo que pida caracteres e imprima ‘VOCAL’ 
#si son vocales y ‘NO VOCAL’ en caso contrario, 
#el programa termina cuando se introduce un espacio.

#almacenar caracter
#identificar si es una letra
#si es una letra transformar a minúscula
#si es igual a 'a' o 'e' o 'i' o 'o' o 'u' es vocal
# si no, es no vocal
#si no lo es pedir otro caracter
#si caracter es espacio termina el programa

print("--->Encontrar vocales<---")
caracter = ""
vocal = False
while caracter!=" ":
	caracter = input("Introduce un caracter> ")
	caracter = caracter.lower
	if(caracter=="a"):
		vocal = True
	elif(caracter=="e"):
		vocal = True
	elif(caracter=="i"):
		vocal = True
	elif(caracter=="o"):
		vocal = True
	elif(caracter=="u"):
		vocal = True
	else:
		vocal = False
	if(vocal == True):
		print("VOCAL")
	else:
		print("NO VOCAL")
		

File: Java/Repetitivas/ej3.java
import java.util.Scanner; //Importar la clase Scanner

/**
 * Algoritmo que pida caracteres e imprima ‘VOCAL’ si son vocales 
 * y ‘NO VOCAL’ en caso contrario, el programa termina cuando se introduce un espacio.
 * 
 * @author Alberto Miguel Martinez Jimenez
 */
/**
 * Algoritmo
 * Almacenar caracter
 * transformar caracter a minuscula
 * si es distinto a 'a' o 'e' o 'i'o 'o' o 'u' no es vocal
 * si es igual es vocal
 * si es igual a espacio sale del programa
 */

public class ej3 {
	
	public static void main(String[] args) {
		char caracter;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		do {
			System.out.println("Introduce un caracter> ");
			caracter = scanner.next().toLowerCase().charAt(0);
			if((caracter=='a')||(caracter=='e')||(caracter=='i')||(caracter=='o')||(caracter=='u')) {
				System.out.println("VOCAL");
			}else{
				System.out.println("NO VOCAL");
			}
		}while(caracter!=' ');
		return;
	}

}
